go/arrow/ipc/message.go

```diff
@@ -154,11 +154,18 @@ type messageReader struct {
 
 	refCount int64
 	msg      *Message
+
+	mem memory.Allocator
 }
 
 // NewMessageReader returns a reader that reads messages from an input stream.
-func NewMessageReader(r io.Reader) MessageReader {
-	return &messageReader{r: r, refCount: 1}
+func NewMessageReader(r io.Reader, opts ...Option) MessageReader {
```

Comment:
do we really need to have a full options addition here as opposed to just taking the allocator as a param? Not sure if there's any other configurable options we'd want to be adding in the future, particularly if we're going to ignore any options that aren't the allocator here.

Reply:
I wanted to avoid a breaking change by adding a param. Is that not an issue? Or are you suggesting a new function?

Reply:
You're right about avoiding breaking with the options. Though I personally doubt many people are calling `NewMessageReader` directly, it's really only exported because the `flight` package uses it to sit on top of the ipc readers to provide access to the additional metadata in flight messages.

It's not a big issue either way I think. More of an aesthetic thing where this gives the impression that message reader would work with other options that get ignored like the reader is doing now.

Reply:
The use of the options pattern is a bit confusing in this library, especially the `WithSchema` option, since in a lot of cases it's not optional and the function will panic without it. In my opinion required things should be real parameters.

I'm going to leave this as-is unless you feel strongly about making it a normal parameter and are okay with the breaking change.

Reply:
I agree with you on the `WithSchema` comment as far as for the writer (it's not required for the reader which will read the schema from the stream, and you only use `WithSchema` for asserting that you're reading a specific schema). 

For now we'll leave it, but after the release of v6.0.1 which will properly tag this so that people can upgrade to v6, and people start using the tagged releases, then i'll be comfortable making the breaking change since it would be in a tagged *major release* that would make sense to allow a breaking change in.